### ISSUE TITLE
Add various crosses endpoints

### ISSUE THREAD

**Issue by chaneylc** [plantbreeding/BrAPI#461]
@BrapiCoordinatorSelby  @trife 

I'd like to start this issue to discuss necessary/potential changes for the v2.1 update.

It would be nice to have the following endpoints for the crosses API; although these are fairly low priority:

Projects in a specific program:
`GET /crossingprojects/{programDbId}
`
`GET /crosses/{programDbId} `  or this could be a /search/crosses   
`GET /plannedcrosses/{programDbId}
`
Some less important but 'discussion worthy' endpoints:
For the Cross object, we have crossDbId, what would the ramifications be for replacing this or adding an observationUnitDbId to show that this Cross is actually a new observation unit and not just an event from two parents?
```
Cross:
      allOf:
      - $ref: '#/components/schemas/CrossNewRequest'
      - properties:
          crossDbId:
            description: the unique identifier for a cross
            example: d105fd6f
            type: string
        type: object
```

The ability to get parents:
`GET /parents/ `       returns CrossParent objects for a given project or planned cross
`GET /parents/{parentType}`

I'm not sure if something like this is possible as an endpoint but returning the difference of /plannedcrosses and crosses.
`GET /plannedcrosses/todo/{plannedCrossesDbId}` or {crossingProjectDbId}
and `GET /plannedcrosses/complete/{plannedCrossesDbId} `or {crossingProjectDbId}
Basically, the ability to return the plannedcrosses that have not been crossed yet. I'm open to ideas on this one, it might be too optimized for the spec.

**Comment by trife** [comment]
A lot to unpack here, but I think there would be more value in the ability to link a crossDbId to a seedLot rather than a new observationUnit. If measurements were being taken _on_ that actual seedLot, it would be the observationUnit, but it's basically its own standalone entity once created.

**Comment by BrapiCoordinatorSelby** [comment]
Sorry I'm late to comment here, this one slipped off my radar

1.  Your first point has already been covered by issue "#480". The URL will be `GET /crosses? programDbId=xyz123` but I think the results will be the same as what you are looking for.

2.  Regarding a connection to ObservationUnits, I agree with @trife. As far as I know, a cross is just an event between two parents and produces seeds. With issue "#446" you can now create a seed lot based on a crossDbId. In this case, I think of a cross as sort of a proto-germplasm, it is theoretically a new line, but has not yet been named/ID'd as an official accession yet. Following that logic, I guess an ObservationUnit could be created from a cross too. @chaneylc Would it suit your needs to add `crossDbId` to the ObservationUnit model, so an ObservationUnit can be created from a germplasm OR a cross? 

3.  Regarding parents, I'm not really sure why you need this? Using `GET /crosses? crossingProjectDbId=xyz123` will get you all the cross data for that project, including all the parent entities. The only advantage I could see would be to get an array of parents without any duplicates. I would want to see proof of a serious performance issue before adding `CrossParent` as a primary BrAPI entity. It is designed to be a simple pointer to a germplasm/OU.

4.  Regarding the difference between plannedCrosses and crosses, I agree that your proposal is a little too specific. But I have some general ideas that could be used to solve the same problem. What if we add a `status` field to the plannedCross object to record if it is `"TODO"`, `"DONE"`, or `"SKIPPED`". A "SKIPPED" status would have to be triggered manually, but we could also add `plannedCrossDbId` into the cross object so when the cross is created it can be linked back to the plan that was used, and when that new cross is uploaded, the server can mark the appropriate plans as "DONE" automatically. With all that, we get the URL `GET /plannedcrosses? crossingProjectDbId=xyz123 & status=TODO` and I think that gets the list you are looking for?

**Comment by BrapiCoordinatorSelby** [comment]
@trife @chaneylc 
If my proposals above sounds like it will suit your needs, please let me know and I will implement them in the spec.
